### PR TITLE
Fix Backup and Restore

### PR DESCRIPTION
**Pull request by mmattel**
This is part three implementing MariaDB 10.6.

* Backup and restore got a big revision which is necessary when intending to upgrade to MariaDB 10.6.
* Recover encrypted files is moved from backup to restore
* Text improvments
* Text fixes

Referencing: #4115 (MariaDb 10.6)

Backport to 10.9, 10.8 and 10.7

### CODE REVIEW

File: modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc [hunk 1]
```diff
@@ -1,69 +1,94 @@
...
-. the `data/` directory containing web-server-related files, `owncloud.log` and possibly encryption keys,
-. a fresh ownCloud database dump,
-. and if applicable custom theme files which are usually located in the `apps` or `apps-external` directory (see xref:developer_manual:core/theming.adoc[Theming ownCloud]).
+.  The `config/` directory.
+.  The `data/` directory. +
+Note that the `data/` directory may not only contain user files, but also keys for encryption.
+.  The `apps/` directory. +
+Note that this is only necessary if you are not using the `apps-external/` directory and have added own apps or themes.
+.  The `apps-external/` directory. +
+Note that this is only necessary if available.
```
Comment (phil-davis) [inline modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc:19]:
```suggestion
Note that this is only necessary if it exists and is in use.
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc [hunk 2]
```diff
@@ -1,69 +1,94 @@
...
-If you install your ownCloud server from our
-https://download.owncloud.org/download/repositories/stable/owncloud/[Open
-Build Service] packages (or from distro packages, which we do not
-recommend), *do not backup your ownCloud server files*, which are the
-other files in your `owncloud/` directory such as `core/`, `3rdparty/`,
-`apps/`, `lib/`, and all the rest of the ownCloud files. If you restore
-these files from backup they may not be in sync with the current package
-versions and will fail the code integrity check. This may also cause
-other errors.
+IMPORTANT: In case you have customized user home direcories or a custom location for encryption keys, you have to manually take care backing them up and restoring them to the same location.
```
Comment (phil-davis) [inline modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc:24]:
```suggestion
IMPORTANT: If you have customized user home direcories or a custom location for encryption keys, you have to manually take care of backing them up and restoring them to the same location.
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc [hunk 3]
```diff
@@ -1,69 +1,94 @@
...
 
-If you install ownCloud from the source tarballs, this will not be an
-issue, and you can safely backup your entire ownCloud installation, with
-the exception of your ownCloud database. Databases cannot be copied, instead
-you must use the database tools to make a correct database dump.
+== Prerequisites
 
-To restore your ownCloud installation from backup, see xref:maintenance/backup_and_restore/restore.adoc[Restoring ownCloud].
+To secure your backup process, stop the web server until backup has finished to prevent user access:
```
Comment (phil-davis) [inline modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc:28]:
```suggestion
To ensure a consistent backup, stop the web server until the backup has finished to prevent user access:
```

**Comment by mmattel** [comment]
Just a note,
docker is missing, but my knowledge lacks there...
But we can add that later on too

File: modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc [hunk 4]
```diff
@@ -1,69 +1,94 @@
...
 
-== Backing up the config/ and data/ Directories
+[source,console,subs="attributes+"]
+----
+sudo service apache2 stop
```
Comment (phil-davis) [inline modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc:32]: I guess this is a guide for people that don't already have a big IT infrastructure and backup plans...
Stopping Apache is going to stop all web services that are being served by the system. That might be only ownCloud, but it might be a lot of unrelated stuff.

File: modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc [hunk 5]
```diff
@@ -1,69 +1,94 @@
...
+Tarball Installation::
+. If you have installed ownCloud from a tarball, you can safely backup the entire installation, with the exception of your ownCloud database. Databases cannot be copied, instead you must use the database tools to make a correct backup.
+
+. You may also backup only the directories named above and the database. To avoid issues, you have to use the same tarball version as the ownCloud version when restoring.
 
-Simply copy your `config/` and `data/` folders to a place outside of your
-ownCloud environment. This example uses `rsync` to copy the two
-directories to `/oc-backupdir`:
+Package Installation::
+If you have installed your ownCloud server from our {depr-repo-url}[Open Build Service] packages (or from distro packages, which is depreciated), *do not backup your ownCloud server files*, which are the other files in your `owncloud/` directory such as `core/`, `3rdparty/`, `lib/`, ect. If you restore these files from backup they may not be in sync with the current package versions, will fail the code integrity check and may also cause other errors.
```
Comment (phil-davis) [inline modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc:43]:
```suggestion
If you have installed your ownCloud server from our {depr-repo-url}[Open Build Service] packages (or from distro packages, which is deprecated), *do not backup your ownCloud server files*, which are the other files in your `owncloud/` directory such as `core/`, `3rdparty/`, `lib/`, etc. If you restore these files from backup they may not be in sync with the current package versions, will fail the code integrity check and may also cause other errors.
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc [hunk 6]
```diff
@@ -77,80 +102,35 @@ sqlite3 data/owncloud.db .dump > owncloud-dbbackup_`date +"%Y%m%d"`.bak
...
-
-If you need to restore files from a backup during which encryption was enabled, proceed as follows with caution.
-
-WARNING: This is *not officially supported*. ownCloud officially supports either restoring the full backup or restoring nothing — not restoring individual parts of it.
+== Backup Cron Jobs
 
-* Restore the file from backup.
-* Restore the file's encryption keys from your backup.
-* Run `occ files:scan`, which makes the scanner find it.
+Use this if you want to protect an accidentially deletion of cron entries, plan to restore to a different server like a physical migration or you need to resetup a server from scratch.
```
Comment (phil-davis) [inline modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc:112]:
```suggestion
Use this if you want to protect against an accidental deletion of cron entries, plan to restore to a different server like a physical migration or you need to resetup a server from scratch.
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc [hunk 7]
```diff
@@ -77,80 +102,35 @@ sqlite3 data/owncloud.db .dump > owncloud-dbbackup_`date +"%Y%m%d"`.bak
...
-   the query below and store the value for later reference.
-   The example query assumes the storage was the same and the file was in the same location.
-   If not, you will need to track down where the file was before.
+Take the following tasks to reactivate your ownCloud instance:
 
-   SELECT encrypted
-   FROM oc_filecache
-   WHERE path = 'path/to/the/file/to/restore'
-     AND storage = <numeric_id>
+.Bring back owncloud into narmal operation mode
```
Comment (phil-davis) [inline modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc:125]:
```suggestion
.Bring back ownCloud into normal operation mode
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc [hunk 8]
```diff
@@ -4,49 +4,77 @@
...
-restore your entire ownCloud installation from backup, with the
-exception of your ownCloud database. Databases cannot be copied, instead you
-must use the database tools to make a correct restoration.
+.  The `config/` directory.
+.  The `data/` directory. +
+Note that the `data/` directory may not only contain user files, but also keys for encryption.
+.  The `apps/` directory. +
+Note that this is only necessary if you are not using the `apps-external/` directory and have added own apps or themes.
+.  The `apps-external/` directory. +
+Note that this is only necessary if available.
```
Comment (phil-davis) [inline modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc:17]:
```suggestion
Note that this is only necessary if it exists and is in use.
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc [hunk 9]
```diff
@@ -4,49 +4,77 @@
...
+.  The ownCloud database.
+.  The custom theme files, if you had any. See xref:developer_manual:core/theming.adoc[Theming ownCloud]. +
+Note that theme files are usually located in either the `apps/` or `apps-external/` directory.
 
-After you have completed the restoration, see how to xref:installation/manual_installation/manual_installation.adoc#script-guided-installation[Set Correct Permissions].
+IMPORTANT: In case you have customized user home direcories or a custom location for encryption keys, you have to manually take care backing them up and restoring them to the same location.
```
Comment (phil-davis) [inline modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc:22]:
```suggestion
IMPORTANT: If you have customized user home directories or a custom location for encryption keys, you have to manually take care backing them up and restoring them to the same location.
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc [hunk 10]
```diff
@@ -4,49 +4,77 @@
...
 
-== Restore Directories
+== Prerequisites
 
-Simply copy your `config/` and `data/` folders to your ownCloud
-environment, for example by running the following command from the backup directory:
+To secure your restore process, stop the web server until restore has finished to prevent user access:
```
Comment (phil-davis) [inline modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc:26]:
```suggestion
To secure your restore process, stop the web server until the restore has finished to prevent user access:
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc [hunk 11]
```diff
@@ -63,3 +91,111 @@ sudo sqlite3 data/owncloud.db < owncloud-dbbackup.bak
...
+
+Take the following tasks to reactivate your ownCloud instance:
+
+.Update the systems data-fingerprint after a backup is restored
+[source,console,subs="attributes+"]
+----
+{occ-command-example-prefix} maintenance:data-fingerprint
+----
+
+.Bring back owncloud into narmal operation mode
```
Comment (phil-davis) [inline modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc:182]:
```suggestion
.Bring back ownCloud into normal operation mode
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc [hunk 12]
```diff
@@ -63,3 +91,111 @@ sudo sqlite3 data/owncloud.db < owncloud-dbbackup.bak
...
+
+.Start the web server
+[source,console,subs="attributes+"]
+----
+sudo service apache2 start
+----
+
+=== Restore Cron Jobs
+
+This is only necessary when you accidentially deleted the crontab entries, restore to a different server like when you have a physical migration or you needed to resetup a server from scratch.
```
Comment (phil-davis) [inline modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc:196]:
```suggestion
This is only necessary when you accidentally deleted the crontab entries, restore to a different server like when you have a physical migration or you needed to resetup a server from scratch.
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc (hunk 4 above)
Comment (mmattel) [reply modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc:32]: You are totally right, therefore I have added at the top:

> This document is intended as a guideline, but the way you implement it depends on your setup

Open for suggestions to make this more eye catching

**Comment by phil-davis** [comment]
IMO it will be good for @EParzefall to also review this. And some appropriate person who works with this stuff for real installations. This is critical docs - when we document any detail of backup and restore, the parts that we document need to be really accurate. Restore from backup using this doc needs to really end up with a valid restored ownCloud installation.

**Comment by mmattel** [comment]
I totally agree on this. My proposal would be the following as it is easier for the review process:

* have a language review
* fix eye-catching things
* merge and backport it

Then, as this is rendered properly and visible on the web (the current version is imho crap), any update or improvement or fix can be done much easier and more people can have a look on it. Reading an antora source having a lot of changes is often hard to do. Doing this in steps eases the process a lot.

**Review by phil-davis — APPROVED** [review]
Looks OK to me. @mmattel you can decide if you want to wait for @EParzefall to also review.

File: modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc (hunk 4 above)
Comment (mmattel) [reply modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc:32]: fixed, relaxed the apache part

File: modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc [hunk 13]
```diff
@@ -1,69 +1,94 @@
 = Backing up ownCloud
 :toc: right
+:depr-repo-url: https://download.owncloud.org/download/repositories/stable/owncloud/
 :page-aliases: go/admin-backup.adoc, \
 maintenance/backup.adoc
 
 == Introduction
 
-If you are running ownCloud in a docker container, refer to xref:installation/docker/index.adoc#upgrading-owncloud-on-docker[Upgrading ownCloud on docker].
+Depending how the ownCloud instance has been installed, you may need slightly different steps to do a backup. You may also use diffrent methods as the data directory can be huge. This document is intended as a guideline, but the way you implement it depends on your setup.
```
Comment (EParzefall) [inline modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc:9]:
```suggestion
Depending on how the ownCloud instance has been installed, you may need slightly different steps to do a backup. You may also use different methods as the data directory can be huge. This document is intended as a guideline, but the way you implement it depends on your setup.
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc [hunk 14]
```diff
@@ -1,69 +1,94 @@
...
-When backing up your ownCloud server, you need to copy the following things in your ownCloud folder (by default located under `/var/www/`):
+In any case you need the following components to be backed up:
 
-. the `config/` directory containing the main configuration file `config.php` plus sample configurations and possibly custom configurations,
-. the `data/` directory containing web-server-related files, `owncloud.log` and possibly encryption keys,
-. a fresh ownCloud database dump,
-. and if applicable custom theme files which are usually located in the `apps` or `apps-external` directory (see xref:developer_manual:core/theming.adoc[Theming ownCloud]).
+.  The `config/` directory.
+.  The `data/` directory. +
+Note that the `data/` directory may not only contain user files, but also keys for encryption.
```
Comment (EParzefall) [inline modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc:15]:
```suggestion
Note that the `data/` directory may not only contain user files but also keys for encryption.
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc [hunk 15]
```diff
@@ -1,69 +1,94 @@
...
 :page-aliases: go/admin-backup.adoc, \
 maintenance/backup.adoc
 
 == Introduction
 
-If you are running ownCloud in a docker container, refer to xref:installation/docker/index.adoc#upgrading-owncloud-on-docker[Upgrading ownCloud on docker].
+Depending how the ownCloud instance has been installed, you may need slightly different steps to do a backup. You may also use diffrent methods as the data directory can be huge. This document is intended as a guideline, but the way you implement it depends on your setup.
 
-When backing up your ownCloud server, you need to copy the following things in your ownCloud folder (by default located under `/var/www/`):
+In any case you need the following components to be backed up:
```
Comment (EParzefall) [inline modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc:11]:
```suggestion
In any case, you need the following components to be backed up:
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc [hunk 16]
```diff
@@ -1,69 +1,94 @@
...
+IMPORTANT: If you have customized user home direcories or a custom location for encryption keys, you have to manually take care of backing them up and restoring them to the same location.
 
-If you install ownCloud from the source tarballs, this will not be an
-issue, and you can safely backup your entire ownCloud installation, with
-the exception of your ownCloud database. Databases cannot be copied, instead
-you must use the database tools to make a correct database dump.
+== Prerequisites
 
-To restore your ownCloud installation from backup, see xref:maintenance/backup_and_restore/restore.adoc[Restoring ownCloud].
+To ensure a consistent backup, stop your webserver to prevent users trying to access ownCloud via the web. As an alternative, you can stop serving the virtual host for ownCloud:
```
Comment (EParzefall) [inline modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc:28]:
```suggestion
To ensure a consistent backup, stop your web server to prevent users from trying to access ownCloud via the web. As an alternative, you can stop serving the virtual host for ownCloud:
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc [hunk 17]
```diff
@@ -1,69 +1,94 @@
...
+----
+sudo service apache2 stop
+----
+
+== Backup Scenarios
+
+Tarball Installation::
+. If you have installed ownCloud from a tarball, you can safely backup the entire installation, with the exception of your ownCloud database. Databases cannot be copied, instead you must use the database tools to make a correct backup.
+
+. You may also backup only the directories named above and the database. To avoid issues, you have to use the same tarball version as the ownCloud version when restoring.
```
Comment (EParzefall) [inline modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc:40]:
```suggestion
. You can also back up only the directories mentioned above and the database. To avoid issues, you have to use the same tarball version as the ownCloud version when restoring.
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc [hunk 18]
```diff
@@ -1,69 +1,94 @@
...
 
-Simply copy your `config/` and `data/` folders to a place outside of your
-ownCloud environment. This example uses `rsync` to copy the two
-directories to `/oc-backupdir`:
+Package Installation::
+If you have installed your ownCloud server from our {depr-repo-url}[Open Build Service] packages (or from distro packages, which is deprecated), *do not backup your ownCloud server files*, which are the other files in your `owncloud/` directory such as `core/`, `3rdparty/`, `lib/`, etc. If you restore these files from backup they may not be in sync with the current package versions, will fail the code integrity check and may also cause other errors.
```
Comment (EParzefall) [inline modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc:43]:
```suggestion
If you have installed your ownCloud server from our {depr-repo-url}[Open Build Service] packages (or from distro packages, which is deprecated), *do not back up your ownCloud server files*, which are the other files in your `owncloud/` directory such as `core/`, `3rdparty/`, `lib/`, etc. If you restore these files from backup they may not be in sync with the current package versions and in that case will fail the code integrity check and may also cause other errors.
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc [hunk 19]
```diff
@@ -1,69 +1,94 @@
...
 
+//missing docker...
+//If you are running ownCloud in a docker container, refer to //xref:installation/docker/index.adoc#upgrading-owncloud-on-docker[Upgrading ownCloud on docker].
+
+=== Backup Directories
+
+If possible, simply copy the directories from your ownCloud installation to your backup location, for example by running the following command from the owncloud directory. The example copies all directories named above:
```
Comment (EParzefall) [inline modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc:50]:
```suggestion
If possible, simply copy the directories from your ownCloud installation to your backup location, for example by running the following command from the owncloud directory. The following example command copies all directories mentioned above:
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc [hunk 20]
```diff
@@ -1,69 +1,94 @@
...
+
+[source,console]
 ----
-rsync -Aax config data /oc-backupdir/
+rsync -Aax config data apps apps-external /oc-backupdir/
 ----
 
-There are many ways to backup normal files. Use whatever
-method you are accustomed to.
+You can also backup the full ownCloud directory which eases the restore as you do not need to install ownCloud first.
```
Comment (EParzefall) [inline modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc:57]:
```suggestion
You can also back up the full ownCloud directory which eases the restore as you do not need to install ownCloud first.
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc [hunk 21]
```diff
@@ -1,69 +1,94 @@
...
+{occ-command-example-prefix} maintenance:mode --on
+----
+
+NOTE: This guide uses a backup file name like `owncloud-dbbackup_<timestamp>.bak`.
 
 === MySQL/MariaDB
 
-MySQL or MariaDB, which is a drop-in MySQL replacement, are the
-recommended database engines. To backup MySQL/MariaDB:
+Depending on the database version and the setup, username and password may not be necessary. The general command to backup MySQL/MariaDB looks like:
```
Comment (EParzefall) [inline modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc:76]:
```suggestion
Depending on the database version and the setup, username and password may not be necessary. The general command to back up MySQL/MariaDB looks like this:
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc [hunk 22]
```diff
@@ -77,80 +102,35 @@ sqlite3 data/owncloud.db .dump > owncloud-dbbackup_`date +"%Y%m%d"`.bak
...
-
-* Restore the file from backup.
-* Restore the file's encryption keys from your backup.
-* Run `occ files:scan`, which makes the scanner find it.
+== Backup Cron Jobs
 
-[NOTE]
-====
-In the DB it will:
+Use this if you want to protect against an accidental deletion of cron entries, plan to restore to a different server like a physical migration or you need to resetup a server from scratch.
```
Comment (EParzefall) [inline modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc:112]:
```suggestion
Use this if you want to protect against an accidental deletion of cron entries, plan to restore to a different server like a physical migration or you need to set up a server from scratch.
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc [hunk 23]
```diff
@@ -77,80 +102,35 @@ sqlite3 data/owncloud.db .dump > owncloud-dbbackup_`date +"%Y%m%d"`.bak
...
-   the query below and store the value for later reference.
-   The example query assumes the storage was the same and the file was in the same location.
-   If not, you will need to track down where the file was before.
+=== Reactivate Your Instance
 
-   SELECT encrypted
-   FROM oc_filecache
-   WHERE path = 'path/to/the/file/to/restore'
-     AND storage = <numeric_id>
+Take the following tasks to reactivate your ownCloud instance:
```
Comment (EParzefall) [inline modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc:123]:
```suggestion
Perform the following tasks to reactivate your ownCloud instance:
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc [hunk 24]
```diff
@@ -77,80 +102,35 @@ sqlite3 data/owncloud.db .dump > owncloud-dbbackup_`date +"%Y%m%d"`.bak
...
+[source,console,subs="attributes+"]
+----
+{occ-command-example-prefix} maintenance:mode --off
+----
 
-   UPDATE oc_filecache
-     SET encrypted = <encrypted>
-     WHERE fileid = <fileid>.
+.Enable browser access
+Start your web server, or alternatively re-enable the virtual host serving ownCloud:
```
Comment (EParzefall) [inline modules/admin_manual/pages/maintenance/backup_and_restore/backup.adoc:132]:
```suggestion
Start your web server, or alternatively enable the virtual host serving ownCloud:
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc [hunk 25]
```diff
@@ -4,49 +4,77 @@
...
-your ownCloud installation. Start with a fresh ownCloud package
-installation in a new, empty directory. Then restore these items from
-your xref:maintenance/backup_and_restore/backup.adoc[Backup]:
+Depending how the ownCloud instance has been installed, you may need slightly different steps to restore it from a backup.
 
-1.  Your `config/` directory.
-2.  Your `data/` directory.
-3.  Your ownCloud database.
-4.  Your custom theme files, if you have any. (See xref:developer_manual:core/theming.adoc[Theming ownCloud])
+In any case you need the following components from your backup:
```
Comment (EParzefall) [inline modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc:9]:
```suggestion
In any case, you need the following components from your backup:
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc [hunk 26]
```diff
@@ -4,49 +4,77 @@
...
+.  The `apps/` directory. +
+Note that this is only necessary if you are not using the `apps-external/` directory and have added own apps or themes.
+.  The `apps-external/` directory. +
+Note that this is only necessary if it exists and is in use.
+.  The ownCloud database.
+.  The custom theme files, if you had any. See xref:developer_manual:core/theming.adoc[Theming ownCloud]. +
+Note that theme files are usually located in either the `apps/` or `apps-external/` directory.
 
-After you have completed the restoration, see how to xref:installation/manual_installation/manual_installation.adoc#script-guided-installation[Set Correct Permissions].
+IMPORTANT: If you have customized user home directories or a custom location for encryption keys, you have to manually take care backing them up and restoring them to the same location.
```
Comment (EParzefall) [inline modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc:22]:
```suggestion
IMPORTANT: If you have customized user home directories or a custom location for encryption keys, you have to manually take care of backing them up and restoring them to the same location.
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc [hunk 27]
```diff
@@ -4,49 +4,77 @@
...
 
-== Restore Directories
+== Prerequisites
 
-Simply copy your `config/` and `data/` folders to your ownCloud
-environment, for example by running the following command from the backup directory:
+To ensure secure your restore process, stop your webserver to prevent users trying to access ownCloud via the web. As an alternative, you can stop serving the virtual host for ownCloud:
```
Comment (EParzefall) [inline modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc:26]:
```suggestion
To ensure a secure restore process, stop your web server to prevent users from accessing ownCloud via the web. As an alternative, you can stop serving the virtual host for ownCloud:
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc [hunk 28]
```diff
@@ -4,49 +4,77 @@
...
+
+Tarball Installation::
+. If you have installed ownCloud from a tarball, you can safely restore the entire installation from the backup, with the exception of your ownCloud database. Databases cannot be copied, instead you must use the database tools to make a correct restoration.
+
+. You may also install a new instance from a tarball and restore the directories named above and the database. To avoid issues, use the same tarball version as the ownCloud version from the backup.
+
+Package Installation::
+If you have installed ownCloud from packages, start with a fresh ownCloud package installation in a new, empty directory. Then restore the above items from your xref:maintenance/backup_and_restore/backup.adoc[Backup].
++
+NOTE: Only copy those files and folders from the `apps/` backup directory which are NOT present after installing. Do not overwrite items of the `apps/` directory in the new installation. This will avoid a fail of the code integrity check and other errors.
```
Comment (EParzefall) [inline modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc:43]:
```suggestion
NOTE: Only copy those files and folders from the `apps/` backup directory which are NOT present after the installation. Do not overwrite items of the `apps/` directory in the new installation. This will prevent a failing code integrity check and other errors.
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc [hunk 29]
```diff
@@ -4,49 +4,77 @@
...
+
+After you have completed restoring files, see how to xref:installation/manual_installation/manual_installation.adoc#script-guided-installation[Set Correct Permissions].
+
+//missing docker...
+
+=== Restore Directories
+
+If possible, simply copy the directories from your backup to your new ownCloud environment, for example by running the following command from the backup directory. The example copies all directories named above:
```
Comment (EParzefall) [inline modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc:51]:
```suggestion
If possible, simply copy the directories from your backup to your new ownCloud environment, for example by running the following command from the backup directory. The following example command copies all directories mentioned above:
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc [hunk 30]
```diff
@@ -63,3 +91,112 @@ sudo sqlite3 data/owncloud.db < owncloud-dbbackup.bak
...
+   the query below and store the value for later reference.
+   The example query assumes the storage was the same and the file was in the same location.
+   If not, you will need to track down where the file was before.
+
+   SELECT encrypted
+   FROM oc_filecache
+   WHERE path = 'path/to/the/file/to/restore'
+     AND storage = <numeric_id>
+
+6. Update the live database instance with retrieved information, by running the
```
Comment (EParzefall) [inline modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc:163]:
```suggestion
6. Update the live database instance with the retrieved information, by running the
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc [hunk 31]
```diff
@@ -63,3 +91,112 @@ sudo sqlite3 data/owncloud.db < owncloud-dbbackup.bak
...
+   following query, substituting the placeholders for the retrieved values:
```
Comment (EParzefall) [inline modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc:164]:
```suggestion
   following query, substituting the placeholders with the retrieved values:
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc [hunk 32]
```diff
@@ -63,3 +91,112 @@ sudo sqlite3 data/owncloud.db < owncloud-dbbackup.bak
...
+
+   UPDATE oc_filecache
+     SET encrypted = <encrypted>
+     WHERE fileid = <fileid>.
+
+== Final Tasks
+
+=== Reactivate Your Instance
+
+Take the following tasks to reactivate your ownCloud instance:
```
Comment (EParzefall) [inline modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc:174]:
```suggestion
Perform the following tasks to reactivate your ownCloud instance:
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc [hunk 33]
```diff
@@ -63,3 +91,112 @@ sudo sqlite3 data/owncloud.db < owncloud-dbbackup.bak
...
+----
+
+.Bring back ownCloud into normal operation mode
+[source,console,subs="attributes+"]
+----
+{occ-command-example-prefix} maintenance:mode --off
+----
+
+.Enable browser access
+Start your web server, or alternatively re-enable the virtual host serving ownCloud:
```
Comment (EParzefall) [inline modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc:189]:
```suggestion
Start your web server, or alternatively enable the virtual host serving ownCloud:
```

File: modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc [hunk 34]
```diff
@@ -63,3 +91,112 @@ sudo sqlite3 data/owncloud.db < owncloud-dbbackup.bak
...
+[source,console]
+----
+sudo service apache2 start
+----
+
+=== Restore Cron Jobs
+
+This is only necessary when you accidentally deleted the crontab entries, restore to a different server like when you have a physical migration or you needed to resetup a server from scratch.
```
Comment (EParzefall) [inline modules/admin_manual/pages/maintenance/backup_and_restore/restore.adoc:197]:
```suggestion
This is only necessary if you accidentally deleted the crontab entries, or you're restoring to a different server to carry out a physical migration or you need to set up a server from scratch.
```

**Review by EParzefall — APPROVED** [review]
Just some language fixes.
